### DISCUSSION TITLE
Slimmer "common" template

### DISCUSSION THREAD

**Discussion by kraenhansen** [callstack/react-native-builder-bob#723] [Ideas]
The [project's readme mentions](https://github.com/callstack/react-native-builder-bob/blob/main/README.md#alternatives):

> The difference is that the generated project with `create-react-native-library` is very opinionated and configured with additional tools.

While I see the value in making a certain choices for its users, I also recognize that it also comes with additional costs on the user to familiarize themselves with these additional tools. Adding to this, as a library developer (admittedly in the advanced end of the spectrum) I have my own preferences on these additional tools. I do however like the idea of initializing the native scaffolding for a React Native library from a CLI and this library has become the de-facto tool for this in the ecosystem, encoding the latest "best practice" around interfacing with React Native core APIs.

Long story short: I would _love_ to be able to run the tool in a mode with less bloat (additional tools that are non-essential to building a React Native library) - and preferably this should be the default (with a choice add additional tools).

Here's a list of dependencies and tools that I would like to strip from the [common](https://github.com/callstack/react-native-builder-bob/tree/main/packages/create-react-native-library/templates/common) template:
- `.github`
- `.yarn` (ideally the choice of package manager should be based on the tool used to initialize the template).
- `.editorconfig`
- `.gitattributes`
- `.nvmrc`
- `.yarnrc.yml`
- `lefthook.yml`
- Dependencies:
  - `@commitlint/config-conventional`
  - `@evilmartians/lefthook`
  - `@release-it/conventional-changelog`
  - `commitlint`
  - `release-it`
  - `turbo`

**Comment by satya164** [comment]
We'll have a template without any non-essential tools in the future.

However, things like adding the choice of package manager puts more maintenance burden. We can't reasonably test so many combinations. So there won't be more customization options than that.

> and preferably this should be the default (with a choice add additional tools)

There's no default. Everything is a choice that you need to select before you generate the project.
